xarray/core/ops.py

```diff
@@ -187,7 +187,16 @@ def array_equiv(arr1, arr2):
     arr1, arr2 = as_like_arrays(arr1, arr2)
     if arr1.shape != arr2.shape:
         return False
-    return bool(((arr1 == arr2) | (isnull(arr1) & isnull(arr2))).all())
+
+    flag_array = (arr1 == arr2)
+    try:
+
+        flag_array |= (isnull(arr1) & isnull(arr2))
+
+    except TypeError:
```

Comment:
could you add a brief comment explaining why we allow `TypeError` here? a reference to this issue (e.g., `GH837`)` could be nice, too.